### PR TITLE
🍰 Create Smartphone Parser

### PR DESCRIPTION
**Pull request by Olderestin**
## What changed?     
- Create smarphone parser using requests and bs4

## Checklist before requesting a review     
- [x] I have performed a self-review of my code
- [x] Code is functional on my machine(Windows 10/Debian GNU/Linux 11 (bullseye))

### CODE REVIEW

File: src/phone/parser/models.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+from mixins import ProcessorCharacteristicsMixin
+from mixins import MemoryCharacteristicsMixin
+from mixins import MainCameraCharacteristicsMixin
+from mixins import FrontCameraCharacteristicsMixin
+from mixins import WirelessTechnologyCharacteristicsMixin
+from mixins import BatteryCharacteristicsMixin
+from mixins import BodyCharacteristicsMixin
+from mixins import PhysicalCharacteristicsMixin
+from mixins import AICharacteristicsMixin
+from mixins import DownloadCharacteristicsMixin
```
Comment (memhard) [inline src/phone/parser/models.py]: why don't use * if you use all classes there

File: src/phone/parser/mixins.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    dimensions_hxwxd: str | None = Field(alias="Габарити (ВхШхГ)", default=None)
+    package_contents: str | None = Field(alias="Комплектація", default=None)
+    legal_information: str | None = Field(alias="Юридична інформація", default=None)
+
+
+class AICharacteristicsMixin(BaseModel):
+    """
+    Mixin class for representing AI characteristics of a smartphone.
+    """
+    # Штучний інтелект
```
Comment (memhard) [inline src/phone/parser/mixins.py]: why comments on uk?

File: src/phone/parser/utils.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+
+def get_characteristics(soup: BeautifulSoup) -> SmartphoneCharacteristics | None:
+    """
+    Extracts the characteristics of the product from the BeautifulSoup object representing a webpage.
+
+    Args:
+        soup: The BeautifulSoup object representing the webpage.
+    """
+    smartphone_characteristics_dict  = {}
```
Comment (memhard) [inline src/phone/parser/utils.py]: bad naming

File: src/phone/parser/utils.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    for div in soup.find_all("div", class_="Characteristic_characteristic__NGJor"):
+        category_data = {span.get_text(strip=True): p.get_text(strip=True) 
+                        for span, p in zip(div.find_all('span'), div.find_all('p'))}
+        smartphone_characteristics_dict.update(category_data)
+
+    try:
+        smartphone_characteristics = SmartphoneCharacteristics.model_validate(smartphone_characteristics_dict)
+    except ValidationError as e:
+        print("ValidationError: ", e)
```
Comment (memhard) [inline src/phone/parser/utils.py]: use loguru for logs

**Review by memhard — CHANGES_REQUESTED** [review]
Need changes!

**Review by Olderestin — COMMENTED** [review]
Changes done.

File: src/phone/parser/utils.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+def get_photo_url(soup: BeautifulSoup) -> list[str]:
+    """
+    Extracts photo URLs from the BeautifulSoup object representing a webpage.
+
+    Args:
+        soup: The BeautifulSoup object representing the webpage.
+
+    Returns:
+        list[str]: List of photo URLs.
```
Comment (memhard) [inline src/phone/parser/utils.py]: we doesn't need to write what type of data we're returning in docstring

File: src/phone/parser/mixins.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    memory_card_support: str | None = Field(alias="Підтримка карток пам'яті", default=None)
+
+
+class MainCameraCharacteristicsMixin(BaseModel):
+    """
+    Mixin class for representing main camera characteristics of a smartphone.
+    
+    Original name of the characteristic on the website:  
+    """
+    # Основна камера
```
Comment (memhard) [inline src/phone/parser/mixins.py]: in class

``` python
class WirelessTechnologyCharacteristicsMixin(BaseModel):
    """
    Mixin class for representing wireless technology characteristics of a smartphone.
    
    Original name of the characteristic on the website: Бездротові технології
    """
    wi_fi: str | None = Field(alias="Wi-Fi", default=None)
    satellite_system: str | None = Field(alias="Супутникова система", default=None)
    bluetooth: str | None = Field(alias="Bluetooth", default=None)
    nfc: str | None = Field(alias="NFC", default=None)
    infrared_port: str | None = Field(alias="Інфрачервоний порт", default=None)
```
there another logic of docstring, why?

**Review by memhard — CHANGES_REQUESTED** [review]
need changes in docs!

**Review by memhard — APPROVED** [review]
LGTM 👍  but need fix docstrings in Returns

**Pull request merged**